packages/jest-emotion/test/react-enzyme.test.js

```diff
@@ -4,29 +4,31 @@ import * as enzyme from 'enzyme'
 import { jsx } from '@emotion/core'
 import { createSerializer as createEnzymeSerializer } from 'enzyme-to-json'
 import { createSerializer } from 'jest-emotion'
+import { toMatchSnapshot } from 'jest-snapshot'
 import React from 'react'
 
-expect.addSnapshotSerializer(createEnzymeSerializer())
-expect.addSnapshotSerializer(createSerializer())
+const createEnzymeSnapshotMatcher = serializerOptions => {
+  const serializer = createEnzymeSerializer(serializerOptions)
+  const identityPrinter = v => v
 
-// Allow deep serialization in tests.
-let useDeepSerializer = false
-const deepEnzymeSerializer = createEnzymeSerializer({ mode: 'deep' })
-expect.addSnapshotSerializer({
-  test: val => useDeepSerializer && deepEnzymeSerializer.test(val),
-  print: deepEnzymeSerializer.print
-})
+  return function(val) {
+    return toMatchSnapshot.call(this, serializer.print(val, identityPrinter))
```

Comment:
My understanding of the serialization was that the printer function (here `identityPrinter`) allows all available serializer plugins (like enzyme and emotion) to serialize children. I don't think `identity` should work here.

A local serializer needs something like this: https://github.com/facebook/jest/blob/bc5c3c7c49f8b190caf060319cc2d88255d9b2ff/packages/jest-snapshot/src/utils.ts#L139-L148

I don't think that's getting any cleaner

Reply:
> My understanding of the serialization was that the printer function (here identityPrinter) allows all available serializer plugins (like enzyme and emotion) to serialize children.

That's my understanding too, but keep in mind that all registered serializers are called in sequence. And my proposed solution does the same - just they don't run as a single sequence, but rather enzyme serializer is gets called first and then I forward its result "manually" to original `toMatchSnapshot` and thus to other registered serializers. So it really behaves pretty much the same.

> I don't think identity should work here.

As you may notice I haven't touched any test file and everything is still passing on my machine after those changes.

> I don't think that's getting any cleaner

 My main argument in favor of the proposed solution is that this makes using deep serializer more explicit, there is no disconnect between enabling it and actually using it.

Reply:
> That's my understanding too, but keep in mind that all registered serializers are called in sequence. And my proposed solution does the same - just they don't run as a single sequence, but rather enzyme serializer is gets called first and then I forward its result "manually" to original toMatchSnapshot and thus to other registered serializers. So it really behaves pretty much the same.

OK, I see that the enzyme serializer only activates on the root wrapper so it does work in this case.

It will break if the enzyme serializer ever changes it's implementation to use nested printer calls or if an enzyme wrapper is not at the root e.g. `expect({wrapper: enzyme.mount(x)}).toMatchShallowSnapshot()`.

Reply:
Seems like forwarding to other printers is really a composition/middleware mechanism, forwarding first and then doing its own operation on the output is rather an uncommon thing to do.

The other concern about not being able to serialize wrapper nested inside a structure is also rather esoteric, I literally don't know when one would like to do this and why it would be a good idea.